service/udp.go

```diff
@@ -84,43 +85,33 @@ func findAccessKeyUDP(clientIP netip.Addr, dst, src []byte, cipherList CipherLis
 
 type packetHandler struct {
 	logger            *slog.Logger
-	natTimeout        time.Duration
 	ciphers           CipherList
-	m                 UDPMetrics
 	ssm               ShadowsocksConnMetrics
 	targetIPValidator onet.TargetIPValidator
-	targetListener    transport.PacketListener
 }
 
+var _ PacketHandler = (*packetHandler)(nil)
+
 // NewPacketHandler creates a PacketHandler
-func NewPacketHandler(natTimeout time.Duration, cipherList CipherList, m UDPMetrics, ssMetrics ShadowsocksConnMetrics) PacketHandler {
-	if m == nil {
-		m = &NoOpUDPMetrics{}
-	}
+func NewPacketHandler(cipherList CipherList, ssMetrics ShadowsocksConnMetrics) PacketHandler {
 	if ssMetrics == nil {
 		ssMetrics = &NoOpShadowsocksConnMetrics{}
 	}
 	return &packetHandler{
 		logger:            noopLogger(),
-		natTimeout:        natTimeout,
 		ciphers:           cipherList,
-		m:                 m,
 		ssm:               ssMetrics,
 		targetIPValidator: onet.RequirePublicIP,
-		targetListener:    MakeTargetUDPListener(0),
 	}
 }
 
-// PacketHandler is a running UDP shadowsocks proxy that can be stopped.
+// PacketHandler is a handler that handles UDP assocations.
 type PacketHandler interface {
+	Handle(pkt []byte, assoc PacketAssociation, lazySlice slicepool.LazySlice)
```

Comment:
Could you add a comment. It's not clear why you need both a `[]byte` and a `LazySlice`.